packages/core/src/state-management/repository.ts

```diff
@@ -325,7 +325,7 @@ export class Repository {
     return new Observable<StreamState>((subscriber) => {
       const id = new StreamID(init.type, init.log[0].cid)
       this.get(id).then((found) => {
-        const state$ = found || new RunningState(init)
+        const state$ = found || new RunningState(init, StateSource.STATESTORE)
```

Comment:
this should also be `StateSource.NETWORK`, although its a little less obvious why here.  If we're making a new RunningState instance here that means that `this.get()` returned null, meaning it couldn't find a RunningState for this stream in either the in-memory cache or the state store. That means this stream must not be pinned.

A question you may have here is - where did the `init` StreamState actually come from?  This `updates$` method is called when an application has a Stream handle in their application code, and then they subscribe to updates to that Stream.  The Stream object itself has a local copy of the StreamState it was initialized with, and that's what's being passed in here. For it not to have been found by `this.get`, the stream must not be pinned, and it must have been long enough since the stream was originally loaded that it has aged out of the `inmemory` cache.  Regardless, since the stream isn't pinned, we can infer that the *original* source of the stream state came from loading it off the network. (*technically* it's possible that it was originally loaded from the state store but has since been unpinned and aged out of the in-memory cache, but that case is still functionally equivalent to if it had never been pinned and had been loaded from the network)

Reply:
thanks for the explanation